src/utils.rs

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate serde_json;
+
+// Macros to help with json serializing of undeclared structs.
+// json_value!({ }) retunns a serde_json::value::Value from an anonymous struct.
+// json!({ }) returns a string from an anonymous struct.
+// Combining both macros let you build arbitrary complex json objects easily.
+//
+// Examples: json!({ type: "light", value: 1.2, on: true });
+//           json!({ type: "notification", value: json_value!({ r: 54, g: 12, b: 88 }) });
+
+#[macro_export]
+macro_rules! json {
+    ({ $($i:ident: $v:expr),* }) => {
+        {
+            use std::collections::BTreeMap;
+            let mut map: BTreeMap<String, serde_json::Value> = BTreeMap::new();
```

Comment:
This will work with "{...}", but not "[...]". Can you make this generic so it takes vectors, too?